spaceR/spaceR/Controllers/LaunchpadViewController.swift

```diff
@@ -63,3 +48,16 @@ extension LaunchpadViewController: UITableViewDelegate, UITableViewDataSource {
         return 200
     }
 }
+
+extension LaunchpadViewController: ViewModelDelegate {
+    
+    func reloadView() {
+        DispatchQueue.main.async {
```

Comment:
Move this line to the repo 😄

Reply:
where in the repo? Because if i don't have it here, i get this purple warning saying "UITableView.reloadData() must be used from main thread only" and then my App crashes when trying to load the rocket data

Reply:
So URLSession pushes us to a background thread so that the service call doesn't happen on the main thread, but once the service call is finished we can move back to the main thread, so essentially why the app is having an issue is because you were trying to update UI from a background thread hence wrapping your UI update in `DispatchQueue.main.async` but ideally we don't want to litter our `ViewControllers` with these types of updates so rather than putting it here put it in your repo on completion (essentially once the service call is finished and you send back your data) that would be a good time to switch back to the main thread. (This goes for API and error 😄 )

Reply:
okay, will have a look into it. Thanks

Reply:
Okay, I put it in the service call because that's where the success completion is returned from. Is that correct?

Reply:
* And the error has been removed as well. All is working as it should😁